mm2src/mm2_main/src/rpc/lp_commands/lp_commands_legacy.rs

```diff
@@ -71,15 +72,16 @@ pub async fn disable_coin(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, St
         Err(err) => return ERR!("!lp_coinfind({}): ", err),
     };
     let coins_ctx = try_s!(CoinsContext::from_ctx(&ctx));
-
     // If a platform coin is to be disabled, we get all the enabled tokens for this platform coin first.
-    let mut tokens_to_disable = coins_ctx.get_tokens_to_disable(&ticker).await;
-    // We then add the platform coin to the list of the coins to be disabled.
-    tokens_to_disable.insert(ticker.clone());
+    let dependent_tokens = coins_ctx.get_dependent_tokens(&ticker).await;
+    if !dependent_tokens.is_empty() {
+        return ERR!("There are tokens dependent on '{}': {:?}", ticker, dependent_tokens);
```

Comment:
Can we use `disable_coin_err` here to return a string that is parsable to `JSON`? It will be a lot easier for the GUI to parse the `dependent_tokens` in this case.
It's also confusing that `disable_coin_err` returns an error as a response but I guess this is needed for backward compatibility. What do you think about this?

Reply:
That makes sense! Done

Yes, we need to maintain the backward compatibility.
> It's also confusing that disable_coin_err returns an error as a response but I guess this is needed for backward compatibility. What do you think about this?